src/unitxt/task.py

```diff
@@ -119,3 +162,94 @@ def process(
 
 class FormTask(Task):
     pass
+
+
+class SubTask(Task):
```

Comment:
Personally, I thinking this kind of multiple inheritance is to complex for users to understand.  Unitxt is already too complicated for most users.   Even in python it's complex, and people will not be able to debug this.   
We are getting a lot of feedback that unitxt becomes too complicated.

I suggest splitting between these changes, the default one is pretty straight forward - and will save people effort (e.g. not specifying text_type in classification, if they don't need to use it in their templates).  

 I think we should discuss the SubTask again - as we already have methods to modify an existing instance using task.qa[metrics=[mymetric]].